controllers/operatorscaler_controller.go

```diff
@@ -0,0 +1,334 @@
+		if err != nil {
+			if !meta.IsNoMatchError(err) {
+				msg := fmt.Sprintf("failed listing %s", resourceMapping.Kind)
+				r.log.Error(err, msg)
+				multierr.AppendInto(&returnErr, err)
+			}
+
+		} else if len(crList.Items) >= 0 {
+
+			for _, pkgName := range resourceMapping.PkgNames {
+				key := fmt.Sprintf(CsvLabelKey, pkgName, r.OperatorNamespace)
+
+				csvList := &oprv1a1.ClusterServiceVersionList{}
+				err = r.Client.List(
+					r.ctx, csvList,
+					client.InNamespace(r.OperatorNamespace),
+					client.MatchingLabels(map[string]string{key: ""}),
+				)
+				if err != nil {
+					r.log.Error(err, "failed listing csvs with label", "label", key)
+					multierr.AppendInto(&returnErr, err)
+				} else {
+					err = r.updateCsvDeplymentsReplicas(csvList.Items...)
+					if err != nil {
+						r.log.Error(err, "failed updating csvs replica")
+						multierr.AppendInto(&returnErr, err)
+					}
+				}
+			}
+		}
+	}
+
+	return returnErr
+}
+
+func (r *OperatorScalerReconciler) updateCsvDeplymentsReplicas(csvList ...oprv1a1.ClusterServiceVersion) error {
+
+	var returnErr error
+
+	for idx := range csvList {
+		csv := &csvList[idx]
+		var updateRequired bool
+		for i := range csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+			deploymentSpec := &csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[i].Spec
+			if *deploymentSpec.Replicas < int32(1) {
+				deploymentSpec.Replicas = ptr.To(int32(1))
+				updateRequired = true
+			}
+		}
+
+		if updateRequired {
+			if err := r.Client.Update(r.ctx, csv); err != nil {
+				r.log.Error(err, "failed updating csv replica", "csvName", csv.Name)
+				multierr.AppendInto(&returnErr, err)
+			} else {
+				r.log.Info("csv updated successfully", "csvName", csv.Name)
+			}
+		}
+	}
+
+	return returnErr
+}
+
+func (r *OperatorScalerReconciler) reconcileDynamicWatchers() error {
+
+	for i := range ResourceMappingList {
+		resourceMapping := ResourceMappingList[i]
+
+		if !r.kindsBeingWatched[resourceMapping.Kind] {
+
+			crd := &extv1.CustomResourceDefinition{}
+			if err := r.Client.Get(r.ctx, types.NamespacedName{Name: resourceMapping.CrdName}, crd); err != nil {
+				if !errors.IsNotFound(err) {
+					r.log.Error(err, "failed getting crd", "crdName", resourceMapping.CrdName)
+					return err
+				}
+			} else {
+				r.log.Info("adding dynamic watch", "kind", resourceMapping.Kind)
+
+				err := r.controller.Watch(
+					source.Kind(
+						r.cache,
+						client.Object(
+							&metav1.PartialObjectMetadata{
+								TypeMeta: metav1.TypeMeta{
+									APIVersion: resourceMapping.ApiVersion,
+									Kind:       resourceMapping.Kind,
+								},
+							},
+						),
+						&handler.EnqueueRequestForObject{},
+						// Trigger the reconcile for creation events of the object.
+						// This ensures the replicas in the CSV are scaled up based on the presence of Custom Resource (CR).
+						createOnlyPredicate,
+						predicate.GenerationChangedPredicate{},
```

Comment:
This predicate is irrelevant if we only consider create events. 
```suggestion
```

Reply:
Done